packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts

```diff
@@ -23,6 +27,107 @@ type UseInitialFormValuesProps = {
+    case "email":
+      return _z.string().optional();
+    case "text":
+      return _z.string().optional();
+    case "textarea":
+      return _z.string().optional();
+    case "multiemail":
+      return _z.union([_z.string(), _z.array(_z.string())]).optional();
+    case "address":
+      return _z.string().optional();
+    case "phone":
+      return _z
+        .string()
+        .regex(/^\+?[0-9]*$/, { message: "Invalid phone number" })
+        .optional();
+    case "number":
+      return _z
+        .string()
+        .refine((val) => !isNaN(Number(val)), { message: "Invalid number" })
+        .transform((val) => Number(val))
+        .optional();
+    case "boolean":
+      return _z
+        .string()
+        .refine((val) => val === "true" || val === "false", { message: "Invalid boolean" })
+        .transform((val) => val === "true")
+        .optional();
+    case "radioInput":
+    case "select":
+    case "radio":
+      if (field.options && field.options.length > 0) {
+        const values = field.options.map((opt) => opt.value) as [string, ...string[]];
+        return _z.union([_z.enum(values), _z.string()]).optional();
+      } else {
+        return _z.any().optional();
+      }
+    case "multiselect":
+    case "checkbox":
+      if (field.options && field.options.length > 0) {
+        const validValues = field.options.map((opt) => opt.value) as [string, ...string[]];
+        return z
+          .union([
+            z
+              .array(z.string())
+              .refine((arr) => arr.some((val) => validValues.includes(val)), {
+                message: "Invalid option in array",
+              })
+              .transform((arr) => arr.filter((val) => validValues.includes(val))),
+
+            z
+              .string()
+              .refine((val) => validValues.includes(val), {
+                message: "Invalid single option",
+              })
+              .optional(),
+          ])
+          .optional();
+      } else {
+        return _z.union([_z.array(_z.string()).optional(), _z.string()]).optional();
+      }
+    // Add more cases as needed
+    default:
+      return _z.any().optional();
+  }
+}
+
+export function checkParseQueryValues(
+  field: Field,
+  parsedQuery: Record<string, unknown> | undefined
+): {
+  success?: boolean;
+  value?: string[] | string | undefined;
+  ignore?: boolean;
+} {
+  if (!parsedQuery || !field) return { ignore: true };
+
+  if (!parsedQuery[field.name]) {
+    return { ignore: true };
+  }
+
+  const schema = getFieldSchema(field);
+  if (!schema) return { ignore: true };
+  const parsedResponses = schema.safeParse(parsedQuery[field.name]);
+  if (parsedResponses.success) {
+    return { success: true, value: parsedResponses.data } as {
+      success: boolean;
+      value: string[] | string | undefined;
+    };
+  } else {
+    return { success: false, value: undefined } as {
+      success: boolean;
+      value: string[] | string | undefined;
+    };
+  }
```

Comment:
I really don't understand why we have to reparse by a new schema. `parsedQuery` is supposed to be already parsed by the response schema